src/main/java/org/jenkinsci/plugins/workflow/flow/FlowExecutionList.java

```diff
@@ -205,8 +208,11 @@ public static class ItemListenerImpl extends ItemListener {
         public void onLoaded() {
             FlowExecutionList list = FlowExecutionList.get();
             for (final FlowExecution e : list) {
-                // The call to FlowExecutionOwner.get in the implementation of iterator() is sufficent to load the Pipeline.
+                // The call to FlowExecutionOwner.get in the implementation of iterator() is sufficient to load the Pipeline.
                 LOGGER.log(Level.FINE, "Eagerly loaded {0}", e);
+                if (e.isComplete()) {
```

Comment:
This will never happen because you are skipping complete executions. There would need to be an alternate iteration method or something.